lms/static/scripts/frontend_apps/components/GradingControls.tsx

```diff
@@ -78,19 +78,16 @@ export default function GradingControls({
   );
 
   useEffect(() => {
-    if (currentStudentIndex >= 0) {
-      changeFocusedUser(students[currentStudentIndex]);
+    if (selectedStudent) {
+      changeFocusedUser(selectedStudent);
     } else {
       changeFocusedUser(null);
     }
-  }, [students, changeFocusedUser, currentStudentIndex]);
-
-  const onSelectStudent = (studentIndex: number) => {
-    setCurrentStudentIndex(studentIndex);
-  };
+  }, [students, changeFocusedUser, selectedStudent]);
 
-  const getCurrentStudent = () => {
-    return currentStudentIndex >= 0 ? students[currentStudentIndex] : null;
+  const onSelectStudent = (lmsId: string | null) => {
```

Comment:
Since `<StudentSelector>` is being passed the list of students, it could also invoke the `onSelectStudent` callback with the specific `StudentInfo` object from that list.